File: Testing/Codeforces/Round 612/B.cpp
#include <bits/stdc++.h>

using namespace std;

using ll = long long;
using ld = long double;

#define sz(x) int((x).size())
#define all(x) begin(x), end(x)

#ifdef LOCAL
    #define eprint(x) cerr << #x << " = " << (x) << endl
    #define eprintf(args...) fprintf(stderr, args), fflush(stderr)
#else
    #define eprint(x)
    #define eprintf(...)
#endif

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n, k;
    cin >> n >> k;
    vector<string> s(n);
    for (int i = 0; i < n; i++)
        cin >> s[i];

    map<string, bool> have;
    for (auto t : s)
        have[t] = true;

    int ans = 0;
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            string toFind;
            for (int p = 0; p < k; p++) {
                if (s[i][p] == s[j][p])
                    toFind.push_back(s[i][p]);
                else
                    toFind.push_back('S' + 'E' + 'T' - s[i][p] - s[j][p]);
            }
            if (have.find(toFind) != have.end())
                ans++;
        }
    }

    cout << (ans / 3) << endl;
}

File: Testing/Codeforces/Round 612/B.py
n, k = map(int, input().split())
s = []
for i in range(n):
    s.append(input())

have = set(s)

ans = 0
for i in range(n):
    for j in range(i + 1, n):
        to_find = []
        for p in range(k):
            if s[i][p] == s[j][p]:
                to_find.append(s[i][p])
            else:
                val = ord('S') + ord('E') + ord('T') - ord(s[i][p]) - ord(s[j][p])
                to_find.append(chr(val))
        if ''.join(to_find) in have:
            ans += 1

print(ans // 3)
